IBX-9813: Cacheable request query parameters (#3033)
ezsystems/developer-documentation: docs/content_management/field_types/field_type_reference/pagefield.md

| `versionNo`    | Version number of the content item to render.                                    |
| `languageCode` | Language code of the content item to render.                                     |

> Optionally, if the layout uses custom block types with a [`cacheable_query_params configuration`](page_blocks.md#block-configuration), pass [`ibexa_append_cacheable_query_params(block)`](page_twig_functions.md#ibexa_append_cacheable_query_params) as the third argument to the [`controller()` Twig function]([[= symfony_doc =]]/reference/twig_reference.html#controller) so that the HTTP cache can vary based on those query parameters.

Reviewer: What about:

```suggestion
If your block needs to be dependent on query parameters like "page" and you already configured your custom block with a [`cacheable_query_params configuration`](page_blocks.md#block-configuration), pass [`ibexa_append_cacheable_query_params(block)`](page_twig_functions.md#ibexa_append_cacheable_query_params) as the third argument to the [`controller()` Twig function]([[= symfony_doc =]]/reference/twig_reference.html#controller) so that the HTTP cache can vary based on those query parameters.
```

if this layout uses custom block types with a [`cacheable_query_params` configuration](page_blocks.md#block-configuration), so it's passed for its HTTP cache to vary on those query parameters.

> Notice that, on a fresh install, no frontend built-in blocks use this feature, but it's only used by [Dashboard blocks]([[= user_doc =]]/getting_started/dashboard/dashboard_block_reference/) My content and Review queue.

Reviewer 2: ```suggestion
Optionally, if the layout uses custom block types with a [`cacheable_query_params configuration`](page_blocks.md#block-configuration), pass [`ibexa_append_cacheable_query_params(block)`](page_twig_functions.md#ibexa_append_cacheable_query_params) as the third argument to the [`controller()` Twig function]([[= symfony_doc =]]/reference/twig_reference.html#controller) so that the HTTP cache can vary based on those query parameters.

Note that in a fresh installation, none of the built-in frontend blocks use this feature; it is currently only used by the [Dashboard blocks[([[= user_doc =]]/getting_started/dashboard/dashboard_block_reference/): "My content" and "Review queue".
```

I like starting with the if, so that the reader can decide if they should continue reading.

This is not a "final" suggestion, feel free to modify it as you wish.

> In a fresh installation, none of the built-in frontend blocks use this feature. It's only used by the [Dashboard blocks]([[= user_doc =]]/getting_started/dashboard/dashboard_block_reference/): "My content" and "Review queue".

Reviewer: > In a fresh installation, none of the built-in frontend blocks use this feature. 

Won't it be a bit misleading? "None of the buit-in" vs "It's only used by" -> maybe we can simply do "In a fresh installation this feature is used by...."

Author: Reworded in b76ddbab9ae4635c94ac22b110f4f27b22bcffbd

| `versionNo`    | Version number of the content item to render.                                    |
| `languageCode` | Language code of the content item to render.                                     |

Use `ibexa_append_cacheable_query_params(block)` as the third parameter of the [`controller()` Twig function]([[= symfony_doc =]]/reference/twig_reference.html#controller)

> so if the block type has a `cacheable_query_params` configuration, it will be passed for its cache to vary on those query parameters.

Author: ```suggestion
so if the block type has a `cacheable_query_params` configuration, it's passed for its cache to vary on those query parameters.
```
